lib/modules/apostrophe-global/index.js

```diff
@@ -139,25 +140,55 @@ module.exports = {
       self.expressMiddleware = self.addGlobalToData;
     };
 
-    // Standard middleware. Fetch the global doc and add it to `req.data` as `req.data.global`.
+    // Fetch the global doc and add it to `req.data` as `req.data.global`, if it
+    // is not already present. If it is already present, skip the
+    // extra query.
+    //
+    // If called with three arguments, acts as middleware.
+    //
+    // If called with two arguments, the first is `req` and the second is
+    // invoked as `callback`.
+    //
+    // If called with one argument, that argument is `req` and a promise
+    // is returned.
 
     self.addGlobalToData = function(req, res, next) {
-      if (self.options.deferWidgetLoading) {
-        req.deferWidgetLoading = true;
+      if (arguments.length >= 3) {
```

Comment:
I think you must've based this branch off another that isn't merged yet? One of mine.

Reply:
i rebased from master? would that do this? Should i copy this out into a fresh from-master branch?